src/pybind/mgr/dashboard/tools.py

```diff
@@ -766,6 +771,18 @@ def str_to_bool(val):
     return bool(strtobool(val))
 
 
+def json_str_to_dict(value):  # type: (AnyStr) -> Dict
+    try:
```

Comment:
Could you please add some documentation. When using doctest as in ``str_to_bool`` you will also provide examples to the developer and you don't need to add additional unittest code.

Reply:
OK, I'll do it. But my understanding is that if you put a descriptive name + type hinting + unit tests you shouldn't need documentation as code should self-descriptive (according to clean code principles). In fact I see more value in tests than comments.
Maybe I didin't achieve the goal of being descriptive enough.

Reply:
Done. Also, updated method name and refactored to address some issues.